lib/Parser/Parse.cpp

```diff
@@ -837,6 +837,14 @@ Symbol* Parser::AddDeclForPid(ParseNodePtr pnode, IdentPtr pid, SymbolType symbo
             }
             break;
         case knopVarDecl:
+
```

Comment:
Function-in-block is essentially a lexical declaration. The module-top-level issue seems equivalent to the issue in a top-level block. Given:
{
    function f(){};
    var f;
}

both Chrome and FF throw redeclaration errors, and we don't. So we do seem to have a general problem with lexical declarations of functions that is not limited to modules. 

The issue also seems similar in spirit to https://github.com/Microsoft/ChakraCore/issues/3275 .